packages/firebase_storage/lib/firebase_storage.dart

```diff
@@ -15,18 +15,50 @@ class FirebaseStorage {
   static FirebaseStorage get instance => new FirebaseStorage();
 
   StorageReference ref() {
-    return const StorageReference._(const <String>[]);
+    return new StorageReference._(const <String>[], this);
   }
 }
 
 class StorageReference {
-  const StorageReference._(this._pathComponents);
+  final FirebaseStorage _firebaseStorage;
+  const StorageReference._(this._pathComponents, this._firebaseStorage);
   final List<String> _pathComponents;
 
+  /// Returns a new instance of [StorageReference] pointing to a child
+  /// location of the current reference.
   StorageReference child(String path) {
     final List<String> childPath = new List<String>.from(_pathComponents)
       ..addAll(path.split("/"));
-    return new StorageReference._(childPath);
+    return new StorageReference._(childPath, _firebaseStorage);
+  }
+
+  ///  Returns a new instance of [StorageReference] pointing to the parent
+  ///  location or null if this instance references the root location.
+  StorageReference getParent() {
+    if(_pathComponents.isEmpty ||
+       _pathComponents.every((String e) => e.isEmpty)) {
+      return null;
+    }
+
+    // Trim for trailing empty path components that can
+    // come from trailing slashes in the path.
+    while(_pathComponents.last.isEmpty) {
+      _pathComponents.removeLast();
+    }
```

Comment:
We should normalize `_pathComponents` where in the `StorageReference` constructor rather than as a side-effect of calling a getter.

Reply:
As discussed, normalization happens on the platform. Changing this to trim the new path list instead.